### PR TITLE
Add Password Generator program to HaikuPorts repository

### PR DESCRIPTION
**Pull request by HaikuUserLinux**

### CODE REVIEW

File: haiku-apps/passwordgenerator/passwordgenerator-1.0.0.recipe [hunk 1]
```diff
@@ -0,0 +1,51 @@
+SUMMARY="A simple password generator app"
+DESCRIPTION="This simple password generator app was originally created on pascal." 
+HOMEPAGE="https://example.com"
+COPYRIGHT="2022 LinuxUser"
+LICENSE="MIT" 
+REVISION="1"
+srcGitRev="54855ece8f2b9d5c9603774a78132b4d7e7fd2fa"
```
Comment (Begasus) [inline haiku-apps/passwordgenerator/passwordgenerator-1.0.0.recipe:7]: Not the latest commit ...

File: haiku-apps/passwordgenerator/passwordgenerator-1.0.0.recipe [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+REVISION="1"
+srcGitRev="54855ece8f2b9d5c9603774a78132b4d7e7fd2fa" 
+SOURCE_URI="https://github.com/HaikuUserLinux/PasswordGeneratorForHaiku/archive/$srcGitRev.tar.gz" 
+CHECKSUM_SHA256="40fbbc8e3253259b4dee9da667973ecf48b7607f124cac9a06fd3a0d58c5fcc1" 
+SOURCE_DIR="PasswordGeneratorForHaiku-$srcGitRev"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+addfix=haiku
+devpref=_devel
```
Comment (Begasus) [inline haiku-apps/passwordgenerator/passwordgenerator-1.0.0.recipe:15]: Why are these 2 lines here?

File: haiku-apps/passwordgenerator/passwordgenerator-1.0.0.recipe [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libQt6Core$secondaryArchSuffix
+	lib:libQt6Gui$secondaryArchSuffix
+	lib:libQt6Widgets$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	$addfix$secondaryArchSuffix$devpref
```
Comment (jmairboeck) [inline haiku-apps/passwordgenerator/passwordgenerator-1.0.0.recipe:29]: this should be
```suggestion
	haiku${secondaryArchSuffix}_devel
```
You need curly braces (`{}`) around secondaryArchSuffix here, then you can remove the 2 variables above that @Begasus asked about.

File: haiku-apps/passwordgenerator/passwordgenerator-1.0.0.recipe [hunk 4]
```diff
@@ -0,0 +1,51 @@
+SUMMARY="A simple password generator app"
+DESCRIPTION="This simple password generator app was originally created on pascal." 
+HOMEPAGE="https://example.com"
```
Comment (Begasus) [inline haiku-apps/passwordgenerator/passwordgenerator-1.0.0.recipe:3]: Update HOMPAGE

File: haiku-apps/passwordgenerator/passwordgenerator-1.0.0.recipe [hunk 5]
```diff
@@ -0,0 +1,51 @@
+SUMMARY="A simple password generator app"
+DESCRIPTION="This simple password generator app was originally created on pascal."
```
Comment (humdingerb) [inline haiku-apps/passwordgenerator/passwordgenerator-1.0.0.recipe:2]: That it was originally a pascal program isn't that important. More interesting would be a short list of its features.

File: haiku-apps/passwordgenerator/passwordgenerator-1.0.0.recipe [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+BUILD()
+{
+	qmake . INSTALL_ROOT=$prefix
+	make $jobArgs
+}
+
+INSTALL()
+{
+	mkdir $appsDir
+	cp passwordgenerator $appsDir
```
Comment (humdingerb) [inline haiku-apps/passwordgenerator/passwordgenerator-1.0.0.recipe:49]: I don't think it's a good idea that the binary's name in $appsDir is different than in Deskbar.
Either "passwordgenerator" or "Password Generator".
Personally, I'd go with something shorter anyway, e.g. "PassGenerator", "PassGen" or something whimsical like "Passwordler".

**Pull request closed without merging**